pubsub/pulsar/pulsar.go

```diff
@@ -104,22 +110,38 @@ func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
 func (p *Pulsar) Subscribe(req pubsub.SubscribeRequest, handler func(msg *pubsub.NewMessage) error) error {
 	channel := make(chan pulsar.ConsumerMessage, 100)
 
-	options := pulsar.ConsumerOptions{
-		Topic:            req.Topic,
-		SubscriptionName: p.metadata.ConsumerID,
-		Type:             pulsar.Failover,
-		MessageChannel:   channel,
-	}
+	if p.metadata.Exclusive == "true" {
+		options := pulsar.ConsumerOptions{
+			Topic:            req.Topic,
+			SubscriptionName: p.metadata.ConsumerID,
+			Type:             pulsar.Exclusive,
+			MessageChannel:   channel,
+		}
 
-	consumer, err := p.client.Subscribe(options)
-	if err != nil {
-		p.logger.Debugf("Could not subscribe %s", req.Topic)
+		options.MessageChannel = channel
+		consumer, err := p.client.Subscribe(options)
+		p.logger.Debugf("This is exclusive subscription type")
+		if err != nil {
+			p.logger.Debugf("Could not subscribe %s", req.Topic)
+		}
+		go p.listenMessage(consumer, handler)
+	} else {
+		options := pulsar.ConsumerOptions{
```

Comment:
The if-else is very similar. It seems that only the type attribute is different, can you handle only the type attribute i if-else and let the common code be deduped?